scripts/esbuild/utils/typescript-source.ts

```diff
@@ -74,25 +76,6 @@ export async function bundleTypeScriptSource(tsPath: string, opts: BuildOptions)
   const jestTypesciptFilename = join(opts.scriptsBuildDir, 'typescript-modified-for-jest.js');
   await fs.writeFile(jestTypesciptFilename, code);
 
-  // Here we transform the TypeScript source from a commonjs to an ES module.
-  // We do this so that we can add an import from the `@environment` module.
-
-  // trim off the last part that sets module.exports and polyfills globalThis since
-  // we don't want typescript to add itself to module.exports when in a node env
-  const tsEnding = `if (typeof module !== "undefined" && module.exports) { module.exports = ts; }`;
-
-  if (!code.includes(tsEnding)) {
-    throw new Error(`"${tsEnding}" not found`);
-  }
-  const lastEnding = code.lastIndexOf(tsEnding);
-  code = code.slice(0, lastEnding);
-
-  const o: string[] = [];
-  o.push(`// TypeScript ${opts.typescriptVersion}`);
-  o.push(code);
-  o.push(`export default ts;`);
-  code = o.join('\n');
-
```

Comment:
Just curious, what changed so we don't need this anymore?

Reply:
The TypeScript team made changes to make it easier consumable for ESM packages (see [announcement](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#easier-api-consumption-from-ecmascript-modules)). I tested it with an `.mjs` file, e.g.
```ts
import * as foo from './node_modules/typescript/lib/typescript.js'

console.log(foo);
```

And it worked great.

Reply:
Nice!